### PR TITLE
chore: Upload snapshots to s3

### PR DESCRIPTION
**Pull request by nahsi**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -33,6 +33,10 @@ jobs:
             target: aarch64-unknown-linux-gnu
             runner: linux-arm64-c-2xlarge
 
+    permissions:
+      contents: read
+      id-token: write
```
Comment (folex) [inline .github/workflows/build.yml:38]: Can you add a comment for that block? no idea what it means

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -90,3 +94,27 @@ jobs:
...
+          method: jwt
+          jwtGithubAudience: "https://github.com/fluencelabs"
+          jwtTtl: 300
+          secrets: |
+            kv/ci/fluence-artifacts id | AWS_ACCESS_KEY_ID ;
+            kv/ci/fluence-artifacts secret | AWS_SECRET_ACCESS_KEY
+
+      - name: Rename nox binary
+        run: |
+          mkdir s3
```
Comment (folex) [inline .github/workflows/build.yml:113]:
```suggestion
          mkdir -p s3
```
so it doesn't fail even if dir already exists?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -90,3 +94,27 @@ jobs:
...
+          cp target/${{ matrix.target }}/${{ steps.profile.outputs.profile }}/nox s3/nox-${{ matrix.arch }}
```
Comment (folex) [inline .github/workflows/build.yml:114]:
```suggestion
          cp "target/${{ matrix.target }}/${{ steps.profile.outputs.profile }}/nox" "s3/nox-${{ matrix.arch }}"
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -33,6 +33,10 @@ jobs:
             target: aarch64-unknown-linux-gnu
             runner: linux-arm64-c-2xlarge
 
+    permissions:
```
Comment (nahsi) [inline .github/workflows/build.yml:36]:
```suggestion
    # requirement for vault-action
    permissions:
```

**Review by folex — APPROVED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (nahsi) [reply .github/workflows/build.yml:115]: No need really cause no bash spaces or bash variables there

**Comment by folex** [comment]
Will it print links into the Summary tab?

**Comment by nahsi** [comment]
> Will it print links into the Summary tab?

Summary already has artifacts uploaded directly

**Pull request merged**